test/extended/router/wildcard.go

```diff
@@ -0,0 +1,105 @@
+package router
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
+	defer g.GinkgoRecover()
+	var (
+		fixture = exutil.FixturePath("testdata", "reencrypt-serving-cert.yaml")
+		oc      *exutil.CLI
+
+		ns string
+	)
+
+	oc = exutil.NewCLI("router-reencrypt", exutil.KubeConfigPath())
+
+	g.BeforeEach(func() {
+		_, routerNs, err := exutil.GetRouterPodTemplate(oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "couldn't find default router")
+
+		routerSvc, err := oc.AdminKubeClient().CoreV1().Services(routerNs).Get("router-default", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred(), "expected %s/%s to exist", routerNs, "router-default")
+
+		if routerSvc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+			g.Skip("default router is not exposed by a load balancer service; skipping wildcard DNS test")
+		}
+
+		ns = oc.KubeFramework().Namespace.Name
+	})
+
+	g.Describe("The default ClusterIngress", func() {
+		g.It("should support default wildcard reencrypt routes through external DNS", func() {
+			execPodName := exutil.CreateExecPodOrFail(oc.AdminKubeClient().Core(), ns, "execpod")
+			defer func() { oc.AdminKubeClient().Core().Pods(ns).Delete(execPodName, metav1.NewDeleteOptions(1)) }()
+
+			g.By(fmt.Sprintf("deploying a service using a reencrypt route using only defaults"))
+			err := oc.Run("create").Args("-f", fixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var hostname string
+			err = wait.Poll(time.Second, changeTimeoutSeconds*time.Second, func() (bool, error) {
+				route, err := oc.RouteClient().Route().Routes(ns).Get("serving-cert", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if len(route.Status.Ingress) == 0 || len(route.Status.Ingress[0].Host) == 0 {
+					return false, nil
+				}
+				hostname = route.Status.Ingress[0].Host
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			url := "https://" + hostname
+			g.By(fmt.Sprintf("verifying the route serves 200 from %s", url))
+			err = waitForURLOK(ns, execPodName, url, changeTimeoutSeconds)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+})
+
+func waitForURLOK(ns, execPodName, url string, timeoutSeconds int) error {
+	cmd := fmt.Sprintf(`
```

Comment:
Is there a reason you didn't reuse the other helpers?

Reply:
All the other ones take/use a Host header, which I don't want in these tests